python/doc/examples/probabilistic_modeling/random_vectors/plot_ratioOfUniforms_exploitation.py

```diff
@@ -0,0 +1,337 @@
+# in the :math:`(x,y)`-space.
+g_XY = draw_RatioOfUniforms_XY(ratioAlgo, 1001)
+g_XY.setTitle(r'$r = $%.3g' % ratioAlgo.getR() + r', $\tau = $%.3g' % ratioAlgo.getAcceptanceRatio())
+
+grid_default = ot.GridLayout(1, 2)
+grid_default.setGraph(0, 0, g_UV)
+grid_default.setGraph(0, 1, g_XY)
+grid_default.setTitle(r'$\log f(x) = \log(\cos(x)) + x$')
+view = otv.View(grid_default)
+
+# %%
+# We change the :math:`r` value and we draw the same plots.
+grid_full = ot.GridLayout(4, 2)
+for i, r in enumerate([0.5, 0.8, 1.2, 2.0]):
+    ratioAlgo.setR(r)
+    g_UV = draw_RatioOfUniforms_UV(ratioAlgo)
+    g_UV.setLegendPosition('')
+    g_UV.setTitle(r'$r = $%.3g' % ratioAlgo.getR() + r', $\tau = $%.3g' % ratioAlgo.getAcceptanceRatio())
+
+    g_XY = draw_RatioOfUniforms_XY(ratioAlgo, 1001)
+    g_XY.setLegendPosition('')
+    g_XY.setTitle(r'$\log f(x) = \log(\cos(x)) + x$, $r = $%.3g' % ratioAlgo.getR() + r', $\tau = $%.3g' % ratioAlgo.getAcceptanceRatio())
+
+    grid_full.setGraph(i, 0, g_UV)
+    grid_full.setGraph(i, 1, g_XY)
+
+grid_full.setTitle(r'$\log f(x) = \log(\cos(x)) + x$')
+view = otv.View(grid_full)
+
+# %%
+# We study the acceptance ratio :math:`\tau(r)` as a function of the parameter :math:`r` for some distributions.
+# We draw the function :math:`r \rightarrow \tau(r)`. We conclude that the default value :math:`r=1` seems to be correct
+# in all these cases.
+#
+# Except for the last distribution, we limit the number of starting points selected for the optimization algorithm.
+mixt_gaus = ot.Mixture([ot.Normal(-2.0, 1.0), ot.Normal(4.0, 1.0)], [0.2, 0.8])
+logPDF_list = [ot.Normal().getLogPDF(), ot.Normal(2).getLogPDF(), ot.Normal(3).getLogPDF(), ot.Exponential().getLogPDF(), mixt_gaus.getLogPDF()]
+range_list = [ot.Normal().getRange(), ot.Normal(2).getRange(), ot.Normal(3).getRange(), ot.Exponential().getRange(), mixt_gaus.getRange()]
+name_list = ['Normal(1)', 'Normal(2)', 'Normal(3)', 'Exponential()', 'Mixture Normal']
+
+g = ot.Graph(r'Acceptance ratio $\tau$ as a function of $r$', 'r', r'$\tau$', True, 'topright')
+g.setLogScale(1)
+numberR = 50
+a = (1e2) ** (2.0 / numberR)
+data_r_ratio = ot.Sample(numberR + 1, 2)
+for k in range(len(logPDF_list)):
+    ratioAlgo = otexp.RatioOfUniforms(logPDF_list[k], range_list[k])
+    for i in range(numberR + 1):
+        r = a ** (i - 0.5 * numberR)
+        ratioAlgo.setR(r)
+        acceptRatio = ratioAlgo.getAcceptanceRatio()
+        data_r_ratio[i] = [r, acceptRatio]
+    draw = ot.Curve(data_r_ratio)
+    draw.setLegend(name_list[k])
+    g.add(draw)
+
+ratioAlgo = otexp.RatioOfUniforms(log_UnscaledPDF, rangePDF)
+for i in range(numberR + 1):
+    r = a ** (i - 0.5 * numberR)
+    ratioAlgo.setR(r)
+    acceptRatio = ratioAlgo.getAcceptanceRatio()
+    data_r_ratio[i] = [r, acceptRatio]
+draw = ot.Curve(data_r_ratio)
+draw.setLegend(r'$\log f(x) = \log(cos(x)) + x$')
+g.add(draw)
+
+g.setLegendPosition("upper left")
+g.setLegendCorner([1.0, 1.0])
+view = otv.View(g)
+
+# %%
```

Comment:
one must call ResourceMap.Reset since you're modifying the resourcemap at the top of the example